base_decorator.rb

```diff
@@ -0,0 +1,10 @@
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
```

Comment:
- [ ] I think it would be great if you add `attr_accessor :nameable` to the namable this is important to put for this methos set and get the value so kindly add it here 💪🏻✅🔥💯💯💯